File: spring/boot/security/with_jwt/layered/src/main/java/org/example/demo/entities/Product.java
package org.example.demo.entities;

import javax.persistence.*;
import java.util.Objects;

@NamedQuery(name="Product.findByNameInAscending",query="from Product where name=:name order by price")
@Entity
public class Product {
    @GeneratedValue
    @Id
    private Long id;
    @Column(unique = true)
    private String name;
    private double price;

    public Product(){}
    public Product(String name, double price){
        this.name=name;
        this.price=price;
    }
    public void setId(Long id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "Product{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", price=" + price +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Product product = (Product) o;
        return Objects.equals(id, product.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: frontend/angular/http-demo/src/app/product.ts
export class Product {
  private _price: number = -1;
  private _name: string = "";
  private _id:number=-1;
  constructor(id:number,name: string, price: number) {
    this._price = price;
    this._name = name;
  }

  get price(): number {
    return this._price;
  }
  set price(_price: number) {
    this._price = _price;
  }

  get name(): string {
    return this._name;
  }
  set name(name: string) {
    this._name = name;
  }

  get id():number{
    return this._id;
  }
  set id(id){
    this._id=id;
  }

}
